src/aggregator/client/queue.go

```diff
@@ -142,10 +147,27 @@ func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
 		bufCh:    make(chan protobuf.Buffer, queueSize),
 		doneCh:   make(chan struct{}),
 	}
+
+	var (
+		ticker *time.Ticker
+		tC     <-chan (time.Time)
+	)
+	if batchSize > 0 && minWriteInterval != 0 {
+		q.buf = make([]byte, 0, batchSize)
+		q.writeInterval = minWriteInterval
+		q.batchSize = batchSize
+		ticker = time.NewTicker(minWriteInterval)
+		tC = ticker.C
```

Comment:
Hm, seems better to pass `minWriteInterval` to `drain` function itself and let it do the ticker creation and stop itself.

i.e.

```go
	if batchSize > 0 && minWriteInterval != 0 {
		q.buf = make([]byte, 0, batchSize)
		q.writeInterval = minWriteInterval
		q.batchSize = batchSize
	}
	q.writeFn = q.conn.Write

	q.wg.Add(2)

	go q.drain(minWriteInterval)

	// ....


func (q *queue) drain(minWriteInterval time.Duration) {
	var tickCh <-chan (time.Time)
	if minWriteInterval >= 0 {
		ticker := time.NewTicker(minWriteInterval)
		defer ticker.Close()
		tickCh = ticker.C
	}

	// as per normal...
}
		
```